### ISSUE TITLE
Conflict with Create0.5

### ISSUE THREAD

**Issue by Dr-WeiAL** [ShetiPhian/Localizations-and-Issues#359]
Here is roughly the information needed
https://github.com/Fabricators-of-Create/Create/issues/425

**Comment by ShetiPhian** [comment]
Where did you get 0.5 from?
The latest build I could find was 0.4.1, and it loads all the way into a world just fine (didn't test past that)

As for the crash; while the log not giving useful names or correct line numbers is normal, it sure doesn't help.
I think the line from create listed is "Minecraft.getInstance().getItemRenderer().renderGuiItem(ICON, x + 2, y + 2);"
So the crash happens in ItemRenderer but I have no clue what "$$6" is or why its null

None of my Mixins touch that class either.
I do have a class that extends it, but never replace the original refence. So its not interfering.

Might be something in Myron as it adds support for items to use obj models. (and it has caused other issues)
Unzip ShetiPhianCore, open the META-INF folder, open the jars folder, extract Myron and see if the crash happens with Myron but without my mods.

**Comment by Dr-WeiAL** [comment]
oh... it looks like myron is no longer maintained?

**Comment by ShetiPhian** [comment]
More or less;
The original author doesn't work on it, and a contributor only updates it to the next version, and even that needs to wait on the author to accept.

Its holding up my 1.19 builds too.

**Comment by Dr-WeiAL** [comment]
So is this problem unsolvable? Or let create solve the compatibility with Myron?

**Comment by ShetiPhian** [comment]
Create may be able to solve/work around it, as 0.4 works and Myron is open source.

It will take much longer for me as I'd need to find/code an alternative (to which rendering is my bane)

**Comment by Dr-WeiAL** [comment]
Okay, I'll explain it to Create.

**Comment by Footage2-Amply-Pounce** [comment]
Hey, ShetiPhian, I was wondering, now that you aren't bundling ShetiPhianCore with Myron anymore, the issue obviously doesn't happen. However, I'm also seeing no issues with any of your mods, so my question is - do we still need to download and use it?

**Comment by ShetiPhian** [comment]
There is a class in ShetiPhianCore that will crash if Myron is missing (actually I think that is wrapped in an error handler) either way that code should be unreachable due to the obj models not getting loaded in the first place.

I think the only thing that will be broken is the ramps in Platforms will have missing model parts. I've done some work on converting them, and will finish as I have time to do so.

**Comment by Tempest-5ky** [comment]
Just confirming: once you've finished converting Platforms, will Platforms no longer require shetiphiancore? Or do you mean that shetiphiancore will no longer need Myron, so once you finish converting the ramps there'll be a myron-less release of shetiphian that will therefore work with create?

**Comment by Footage2-Amply-Pounce** [comment]
> ... shetiphiancore will no longer need Myron, so once you finish converting the ramps there'll be a myron-less release of shetiphian that will therefore work with create?

ShetiPhianCore on 1.19+ currently doesn't come bundled with Myron already, so if Create was on 1.19, the issue shouldn't be there. But Platforms relies on some part of Myron, so if you don't have it installed expect some issues like missing model parts (I so far haven't noticed any, but I also didn't extensively test it). Later there will be a version of ShetiPhianCore/Platforms that doesn't need Myron at all.

**Comment by Tempest-5ky** [comment]
I see. Do you know if there'll be a release of either of those with their respective fixes for 1.18.2, because of Create currently being in that version?

**Comment by itsdinkd** [comment]
> I see. Do you know if there'll be a release of either of those with their respective fixes for 1.18.2, because of Create currently being in that version?

I second this question. Would be very helpful as your mods are a must in my pack :)

**Comment by ardissaps** [comment]
deleted myron jar from shetipiancore jar, the game launch no problem only few missing model, so my gues is it can be fixed with some model editing maybe?
[![image](https://user-images.githubusercontent.com/82106625/184531181-b1952d27-cad0-4e95-a1f6-b51ab0c89041.png)](https://github.com/ShetiPhian/Localizations-and-Issues/issues/372)

**Comment by ardissaps** [comment]
deleted myron jar from shetipiancore jar, the game launch no problem only few missing model, so my gues is it can be fixed with some model editing maybe?
[![image](https://user-images.githubusercontent.com/82106625/184531181-b1952d27-cad0-4e95-a1f6-b51ab0c89041.png)](https://github.com/ShetiPhian/Localizations-and-Issues/issues/372)

**Comment by Tempest-5ky** [comment]
> deleted myron jar from shetipiancore jar (1.18.2), the game launch no problem only few missing model, so my gues is it can be fixed with some model editing maybe? (hand rail when on stairs, and ramp) 

Could you send me the modified jar for that? I really need to get my server up again and i'm fine with missing models if it means it _works_ for now

**Comment by ShetiPhian** [comment]
I see my mobile response didn't get posted (guess you can't remove the quoted part) but here it is:

> Platforms will still need ShetiPhianCore, but SPC will no longer need/use/include Myron.
> 
> Model changes will be backported to 1.17 and 1.18
> 
> Current builds of SPC can be altered to remove Myron. Open the jar in a zip program like 7zip, enter the meta-inf folder and delete the jar-in-jar folder (think that's the correct name, Myron is inside the folder)

Changes since then:
- I'll also backport to 1.16
- The folder is "jars" not "jar-in-jar"
- Since I have time right now, I'm going to work on builds of ShetiPhianCore that completely removes Myron (not just the jar but the supporting code). They should be on Curse later today.

**Comment by Tempest-5ky** [comment]
Thank you so much for the update! This will help out a ton of people, we really appreciate it :D

**Comment by ShetiPhian** [comment]
Currently waiting for approval but here is the links:

1.16 -- https://www.curseforge.com/minecraft/mc-mods/shetiphiancore-fabric/files/3934930
1.17 -- https://www.curseforge.com/minecraft/mc-mods/shetiphiancore-fabric/files/3934932
1.18 -- https://www.curseforge.com/minecraft/mc-mods/shetiphiancore-fabric/files/3934933

**Comment by ShetiPhian** [comment]
Platforms will still need ShetiPhianCore, but SPC will no longer
need/use/include Myron.

Model changes will be backported to 1.17 and 1.18

Current builds of SPC can be altered to remove Myron. Open the jar in a zip
program like 7zip, enter the meta-inf folder and delete the jar-in-jar
folder (think that's the correct name, Myron is inside the folder)

**Comment by ardissaps** [comment]
> Platforms will still need ShetiPhianCore, but SPC will no longer need/use/include Myron. Model changes will be backported to 1.17 and 1.18 Current builds of SPC can be altered to remove Myron. Open the jar in a zip program like 7zip, enter the meta-inf folder and delete the jar-in-jar folder (think that's the correct name, Myron is inside the folder)

has the ramp model converted?

**Comment by ShetiPhian** [comment]
The main parts have been, just have a few subparts left.
It's been that way for a while now though, issue is other then paying bills
I haven't used my PC for a couple of months.
When I have the time to I'll compile what I have and get a new build out.
Might not be until the end of the month or start of the next though.

**Comment by ShetiPhian** [comment]
Builds are up, and ramps are visible again without Myron.

I'm going to close this issue now as it was for a conflict with Myron, and any model problems should be in its own issue going forward.
